libsolidity/ast/Types.h

```diff
@@ -748,6 +748,7 @@ class StructType: public ReferenceType
 	{
 		return location() == DataLocation::Storage ? std::make_shared<IntegerType>(256) : shared_from_this();
 	}
+	virtual TypePointer decodingType() const override;
```

Comment:
`encodingType` above seems to be the exact same? Also if two functions must exists it is probably better placing them next to each other, either header or code.

Reply:
They are different for some types.

Reply:
Ah right, but removed it from struct because the default is to use the encoding type.